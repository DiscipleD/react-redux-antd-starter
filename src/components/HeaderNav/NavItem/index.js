/**
 * Created by jack on 16-9-23.
 */

import React from 'react';
import { Link } from 'react-router';

class NavItem extends React.Component {
	render() {
		const item = this.props.item;
		const navItemElement = item.link
			? <Link to={item.link}>{item.label}</Link>
				: typeof item.fn === 'function'
				? <a onClick={item.fn}>{item.label}</a>
				: item.label;
		return (
			<li className="nav-item">{navItemElement}</li>
		)
	}
}

export default NavItem;
