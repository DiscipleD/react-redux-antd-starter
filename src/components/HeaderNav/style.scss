.header-nav {
  padding: 0 50px;
  background: #404040;
  height: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;

  .header-nav-logo {
    text-align: center;
    padding: 2px 6px;
    font-size: 18px;
    color: #fff;
    width: 120px;
    height: 32px;
    background: #333;
    border-radius: 6px;
  }

  .user-actions{
  }
}
