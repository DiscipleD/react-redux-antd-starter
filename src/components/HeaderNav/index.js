/**
 * Created by jack on 16-9-6.
 */

import React from 'react';

import './style.scss';
import NavItem from './NavItem';

class HeaderNav extends React.Component {
	render() {
		const { logo, navItems: navItemsSetting } = this.props;
		const navItems = (navItemsSetting || []).map((item, key) =>
			<NavItem item={item} key={key} />
		);
		return (
			<div className="header-nav">
				<div className="header-nav-logo" onClick={logo.onClick}>{logo.label}</div>
				<ul className="user-actions">
					{navItems}
				</ul>
			</div>
		)
	}
}

export default HeaderNav;
