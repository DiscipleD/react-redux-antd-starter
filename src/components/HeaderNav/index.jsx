/**
 * Created by jack on 16-9-6.
 */

import React from 'react';
import {Link} from 'react-router';
import {Menu} from 'antd';

import './style.scss';
export default class HeaderNav extends React.Component {
	render() {
		return (
			<div className="header-nav">
				<div className="header-nav-wrapper">
					<div className="header-nav-logo">Project Logo</div>
					<Menu
						theme="dark"
						mode="horizontal"
						defaultSelectedKeys={[this.props.path === '/' ? '/a' : this.props.path]}
						style={{lineHeight: '64px'}}
					>
						<Menu.Item key="/a">
							<Link to="/a">Page A</Link>
						</Menu.Item>
						<Menu.Item key="/b">
							<Link to="/b">Page B</Link>
						</Menu.Item>
					</Menu>
				</div>
			</div>
		)
	}
}
