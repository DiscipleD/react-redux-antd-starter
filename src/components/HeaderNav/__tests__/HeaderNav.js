/**
 * Created by jack on 16-9-25.
 */

import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-addons-test-utils';

import HeaderNav from '../index';

describe('HeaderNav Component', () => {
	const logo = {
		name: 'logo',
		label: 'logo',
		onClick: jest.fn()
	};
	const navItems = [{
		name: 'userName',
		label: 'aa',
		link: '/a'
	}, {
		name: 'logout',
		label: 'Logout',
		fn: jest.fn()
	}];
	const headerNav = <HeaderNav logo={logo} navItems={navItems}/>;
	const headerNavElement = ReactTestUtils.renderIntoDocument(headerNav);
	const headerNavNode = ReactDOM.findDOMNode(headerNavElement);

	it('Logo', () => {
		expect(headerNavNode.querySelector('.header-nav-logo').textContent).toBe(logo.label);

		ReactTestUtils.Simulate.click(headerNavNode.querySelector('.header-nav-logo'));
		expect(logo.onClick).toBeCalled();
	});

	it('Nav items render correct', () => {
		expect(headerNavNode.querySelectorAll('.nav-item').length).toBe(navItems.length);
	});

	it('0 nav item.', () => {
		const logo = {};
		const headerNav = ReactTestUtils.renderIntoDocument(
			<HeaderNav logo={logo}/>
		);
		const headerNavNode = ReactDOM.findDOMNode(headerNav);

		expect(headerNavNode.querySelectorAll('.nav-item').length).toBe(0);
	});

	it('match snapshot', () => {
		const headerNavSnapshot = renderer.create(headerNav).toJSON();
		expect(headerNavSnapshot).toMatchSnapshot();
	})
});
