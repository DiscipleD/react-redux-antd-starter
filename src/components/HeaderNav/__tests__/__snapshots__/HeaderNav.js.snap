exports[`HeaderNav Component match snapshot 1`] = `
<div
  className="header-nav">
  <div
    className="header-nav-logo"
    onClick={[Function mockConstructor]}>
    logo
  </div>
  <ul
    className="user-actions">
    <li
      className="nav-item">
      <a
        onClick={[Function bound handleClick]}
        style={Object {}}>
        aa
      </a>
    </li>
    <li
      className="nav-item">
      <a
        onClick={[Function mockConstructor]}>
        Logout
      </a>
    </li>
  </ul>
</div>
`;
